app/scripts/controllers/serviceInstance.js

```diff
@@ -128,33 +129,52 @@ angular.module('openshiftConsole')
       updateParameterData();
     };
 
-    var updateServiceClass = function() {
-      // If we've previously loaded the service class or a request is in flight, don't do anything.
-      if (!$scope.serviceInstance || $scope.serviceClass || serviceClassPromise) {
+    var updateServicePlan = function() {
+      var servicePlanName = _.get($scope.serviceInstance, 'spec.clusterServicePlanRef.name');
+      $scope.plan = _.find($scope.servicePlans, function(plan) {
+        return _.get(plan, 'metadata.name') ===  servicePlanName;
+      });
+      updateParameterSchema();
+      updateEditable();
+    };
+
+    var updateServicePlans = function() {
+      if (!$scope.serviceClass || servicePlansPromise) {
         return;
       }
 
-      serviceClassPromise = ServiceInstancesService.fetchServiceClassForInstance($scope.serviceInstance).then(function (serviceClass) {
-        $scope.serviceClass = serviceClass;
-        $scope.displayName = serviceInstanceDisplayName($scope.serviceInstance, $scope.serviceClass);
-
-        updateBreadcrumbs();
-        serviceClassPromise = null;
-
-        Catalog.getServicePlansForServiceClass($scope.serviceClass).then(function (plans) {
-          plans = plans.by('metadata.name');
-
+      if (!$scope.servicePlans) {
+        servicePlansPromise = Catalog.getServicePlansForServiceClass($scope.serviceClass).then(function (plans) {
           var servicePlanName = _.get($scope.serviceInstance, 'spec.clusterServicePlanRef.name');
-          $scope.servicePlans = _.reject(plans, function(plan) {
+          $scope.servicePlans = _.reject(plans.by('metadata.name'), function(plan) {
             return _.get(plan, 'status.removedFromBrokerCatalog') && (plan.metadata.name !== servicePlanName);
```

Comment:
I just realized if you change the instance plan twice without leaving the page, you'll probably be able to select a removed plan again if it was initially selected. Nothing related to this PR, but we might want to test and open an issue.

Reply:
Yeah, I thought of the same thing but really really really edge case. Can't think of a good fix except to re-request the plans on a change.

Reply:
Check if current instance plan != $scope.plan on instance watch updates and re-filter the existing $scope.servicePlans list?

Reply:
True, I was actually thinking of plans that were removed after the original fetch.